File: leetcode/python/binary-tree-inorder-traversal.py
from typing import Optional, List
class TreeNode:
    def __init__(self,val = 0, left = None, right = None) -> None:
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode])->List[int]:
        ans = []
        def dfs(src: Optional[TreeNode]):
            if src is None:
                return
            dfs(src.left) 
            ans.append(src.val)
            dfs(src.right)
        dfs(root)
        return ans

File: leetcode/java/BinaryTreeInorderTraversal.java
import java.util.ArrayList;
import java.util.List;

public class BinaryTreeInorderTraversal {

    public void recurse(TreeNode root, List<Integer> nodes){
        if(root == null) return;
        recurse(root.left, nodes);
        nodes.add(root.val);
        recurse(root.right, nodes);
    }
    public List<Integer> inorderTraversal(TreeNode root){
        List<Integer> nodes = new ArrayList<>();
        recurse(root,nodes);
        return nodes;
    }
}